### ISSUE TITLE
CLI Argument Parsing is Broken with `--single-thread`

### ISSUE THREAD

**Issue by alexlafroscia** [vitest-dev/vitest#3107]
### Describe the bug

When providing the `--single-thread` argument to the Vitest CLI, the presence of other arguments can mess with the way that the command line arguments are parsed.

This works fine

```
vitest --inspect-brk --single-thread
```

but this gives me a complaint that `--single-thread` is necessary to use `--inspect-brk`, even though it _is_ present already

```
vitest --inspect-brk --single-thread SomeTestModuleNameToFilterOn
```

<img width="600" alt="CleanShot 2023-03-30 at 11 40 58@2x" src="https://user-images.githubusercontent.com/1645881/228890516-2f5abb03-235d-4202-9428-1ecd66c6737d.png">

### Reproduction

https://stackblitz.com/edit/vitest-dev-vitest-brxbu1?file=package.json&initialPath=__vitest__

### System Info

```shell
System:
    OS: Linux 5.0 undefined
    CPU: (8) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
    Memory: 0 Bytes / 0 Bytes
    Shell: 1.0 - /bin/jsh
  Binaries:
    Node: 16.14.2 - /usr/local/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 7.17.0 - /usr/local/bin/npm
  npmPackages:
    @vitest/ui: latest => 0.29.8 
    vite: latest => 4.2.1 
    vitest: latest => 0.29.8
```

### Used Package Manager

npm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by alexlafroscia** [comment]
Interestingly, this _doesn't_ seem to have the same error

```
vitest --single-thread --inspect-brk SomeTestModuleNameToFilterOn
```

But the arguments really shouldn't be order-dependent. The documentation also describes them with `--inspect-brk` coming first.

**Comment by sheremet-va** [comment]
I don't think it's possible to pass it down like that. [`cac`](https://www.npmjs.com/package/cac) reads it as a value for the `--argument`.

**Comment by AriPerkkio** [comment]
The difference of these two comes from here:

https://github.com/vitest-dev/vitest/blob/326b242d8a7798b517c1c8de93321a983fc4ef09/packages/vitest/src/node/config.ts#L109-L114

We are explicitly looking for `resolved.singleThread !== true`, when `singleThread` is actually a filename. When comparing `resolved.inspect` it is coerced to `Boolean`. 

As the described workflow of this issue is expected to be common I think we should explicitly convert these flags to `Boolean` in resolved configuration. It does not require much code changes.

**Comment by alexlafroscia** [comment]
I'm not sure that #3111 addressed this issue in its entirety. It seems that now, additional filters after the arguments to launch Vitest with the debugger enabled are just ignored entirely

If I run the following, just my one `tests/App.test.ts` file is executed

```sh
yarn vitest App.test
```

If I do the same, but with the debugging commands provided, it _does_ launch in debugging mode, but starts running _all_ of my tests

```sh
yarn vitest --inspect-brk --single-thread App.test
```

I would expect that the behavior here is that it launches _just_ `tests/App.test.ts` but with the ability to attach a debugger.

**Comment by AriPerkkio** [comment]
I'm able to reproduce this. It seems that kebab-case vs camelCase produces different results. I wonder if this is a bug in `cac`.

```sh
# Runs only 'math.test.ts' with single thread
$ pnpm run test --singleThread tests/math.test.ts

# Runs all tests in single thread
$ pnpm run test --single-thread tests/math.test.ts 
```

**Comment by AriPerkkio** [comment]
```js
import { cac } from "cac";

const cli = cac("repro");
cli.option("--single-thread", "Testing");

console.log(JSON.stringify(cli.parse(), null, 2));
```

```console
$ node cac-repro.mjs --single-thread tests/file.test.ts
{
  "args": [],
  "options": {
    "--": [],
    "singleThread": "tests/file.test.ts"
  }
}

$ node cac-repro.mjs --singleThread tests/file.test.ts
{
  "args": [
    "tests/file.test.ts"
  ],
  "options": {
    "--": [],
    "singleThread": true
  }
}
```

**Comment by vitkarpov** [comment]
Maybe add "1" or "y" after "single-thread"? Otherwise, `cac` thinks the path is the argument value rather than a path. 

```console
$ node cac-repro.mjs --single-thread 1 tests/file.test.ts
{
  "args": [
    "tests/file.test.ts"
  ],
  "options": {
    "--": [],
    "singleThread": 1
  }
}
```

**Comment by sheremet-va** [comment]
This issue should be fixed in 1.3.0.
